R/repeated_measures_d.R

```diff
@@ -0,0 +1,302 @@
+#'
+#' # d is the same as Cohen's d for two independent groups:
+#' cohens_d(rt ~ cond, data = rouder2016, ci = NULL)
+#'
+#' @export
+repeated_measures_d <- function(x, y,
+                                data = NULL,
+                                mu = 0, method = c("rm", "av", "z", "b", "d", "r"),
+                                adjust = TRUE,
+                                ci = 0.95, alternative = "two.sided",
+                                verbose = TRUE, ...) {
+  method <- match.arg(method)
+  if (.is_htest_of_type(x, "t-test")) {
+    return(effectsize(x, type = paste0("rm_", method), verbose = verbose, adjust = adjust, ...))
+  }
+
+  alternative <- .match.alt(alternative)
+  data <- .get_data_paired(x, y, data = data, method = method, verbose = verbose, ...)
+
+  if (method %in% c("d", "r")) {
+    values <- .replication_d(data, mu = mu, method = method)
+  } else {
+    values <- .paired_d(data, mu = mu, method = method)
+  }
+
+  out <- data.frame(d = values[["d"]])
+
+  if (.test_ci(ci)) {
+    # Add cis
+    out$CI <- ci
+    ci.level <- .adjust_ci(ci, alternative)
+
+    alpha <- 1 - ci.level
+    probs <- c(alpha / 2, 1 - alpha / 2)
+    qs <- stats::qnorm(probs)
+
+    confint <- out[["d"]] + qs * values[["se"]]
+    out$CI_low <- confint[1]
+    out$CI_high <- confint[2]
+
+    ci_method <- list(method = "normal")
+    out <- .limit_ci(out, alternative, -Inf, Inf)
+  } else {
+    ci_method <- alternative <- NULL
+  }
+
+
+  if (adjust) {
+    J <- .J(values[["df"]])
+
+    out[, colnames(out) %in% c("d", "CI_low", "CI_high")] <-
+      out[, colnames(out) %in% c("d", "CI_low", "CI_high")] * J
+
+    attr(out, "table_footer") <- "Adjusted for small sample bias."
+  }
+
+  # rename column to method
+  colnames(out)[1] <- switch(method,
+    d = "Cohens_d",
+    b = "Beckers_d",
+    paste0("d_", method)
+  )
+
+  class(out) <- c("effectsize_difference", "effectsize_table", "see_effectsize_table", class(out))
+  .someattributes(out) <- .nlist(
+    mu, ci, ci_method, alternative,
+    approximate = FALSE
+  )
+  return(out)
+}
+
+#' @rdname repeated_measures_d
+#' @export
+rm_d <- repeated_measures_d
+
+#' @keywords internal
+.paired_d <- function(data, mu, method) {
+  x <- data[["x"]]
+  y <- data[["y"]]
+
+  m <- mean(x - y)
+  n <- length(x)
+  df <- n - 1
+  r <- stats::cor(x, y)
+
+  if (method == "rm") {
+    f <- 2 * (1 - r)
+
+    s <- stats::sd(x - y) / sqrt(f)
+    d <- (m - mu) / s
+
+    # Cooper et al., 2009, eq 12.21
+    se <- sqrt(((1 / n) + (d^2) / (2 * n)) * f)
+  } else if (method == "av") {
+    s <- sqrt((stats::var(x) + stats::var(y)) / 2)
+    d <- (m - mu) / s
+
+    # Algina & Keselman, 2003, eq 4 (from Bird 2002?)
+    se <- sqrt(stats::var(x - y) / (s^2 * n))
+  } else if (method == "z") {
+    s <- stats::sd(x - y)
+    d <- (m - mu) / s
+
+    # Hedges and Olkin, 1985, page 86, eq 14
+    se <- sqrt((1 / n) + (d^2) / (2 * n))
+  } else if (method == "b") {
+    s <- stats::sd(y)
```

Comment:
would it be better if it was `stats::sd(x)`? I think pre-test would likely be the first argument

Reply:
I'm being consistant with `t.test()` where `x - y`, implying that `y` is the control group / pre measure. 
This is also what is done for `glass_delta()`.